src/leds.c

```diff
@@ -169,14 +169,28 @@ static void led_set_color(
     leds->led_data[led] = RGBW(r, g, b, w);
 }
 
-static void strip_set_color(
-    Leds *leds, const LedStrip *strip, uint32_t color, float brightness, float blend
+static void strip_set_color_range(
+    Leds *leds,
+    const LedStrip *strip,
+    uint32_t color,
+    float brightness,
+    float blend,
+    uint8_t idx_start,
+    uint8_t idx_end
 ) {
-    for (uint8_t i = 0; i < strip->length; ++i) {
+    uint8_t start = idx_start < strip->length ? idx_start : 0;
```

Comment:
I don't think you need this calculation, if `start` would be greater than `end`, the for loop below just won't execute at all (programmer error).

Reply:
Good observation, also done!